### PR TITLE
feat(did): get did doc from chain when cache server fails

### PR DESCRIPTION
**Pull request by dwojno**

### CODE REVIEW

File: src/iam.ts [hunk 1]
```diff
@@ -215,11 +215,29 @@ export class IAM extends IAMBase {
...
-        service: didDoc.service as (IServiceEndpoint & ClaimData)[]
-      };
+      try {
+        const didDoc = await this._cacheClient.getDidDocument({ did, includeClaims });
+        return {
+          ...didDoc,
+          service: didDoc.service as (IServiceEndpoint & ClaimData)[]
+        };
+      } catch (err) {
+        if (err?.response?.status === 404) {
```
Comment (jrhender) [inline src/iam.ts:225]: I don't think we could catch only on 404, I think we should catch any error and just fallback to the blockchain. Does that seem reasonable?

File: src/iam.ts [hunk 2]
```diff
@@ -215,11 +215,29 @@ export class IAM extends IAMBase {
...
+              ...document,
+              service: includeClaims
+                ? ((await this.downloadClaims({
+                    services:
+                      document.service && document.service.length > 0 ? document.service : []
+                  })) as (IServiceEndpoint & ClaimData)[])
+                : []
+            };
+          }
+        }
```
Comment (jrhender) [inline src/iam.ts:238]: Instead of duplicating the code here, I would just fall through to the code below, I think it is the same.

**Review by jrhender — CHANGES_REQUESTED** [review]
A couple suggestions to address

File: src/iam.ts (hunk 1 above)
Comment (dwojno) [reply src/iam.ts:225]: Ok, let's fallback to blockchain on every error except 401

File: src/iam.ts (hunk 2 above)
Comment (dwojno) [reply src/iam.ts]: 👍

File: src/iam.ts (hunk 1 above)
Comment (jrhender) [reply src/iam.ts]: I think there's a case to be made for getting from chain even if you get a 401 from the cache-server. To me, it doesn't really sense that iam-client-lib would say "oh, you're not authorized for the cache-server, so I'm not going to let you fallback to the blockchain directly". What do you think?

Comment (dwojno) [reply src/iam.ts]: Yeah, your right 👍  there is no need to be authorized to get something from chain, I'll change that

**Review by manihagh — APPROVED** [review]

File: tsconfig.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "noImplicitReturns": true,
     "noImplicitThis": true,
     "noLib": false,
-    "noUnusedLocals": false,
+    "noUnusedLocals": true,
```
Comment (jrhender) [inline tsconfig.json:21]: 👍

**Review by jrhender — APPROVED** [review]
Nice work!

**Pull request merged**